src/accounts/accounts.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Controller,
+  Post,
+  Body,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { Accounts } from './accounts.model';
+import { AccountsService } from './accounts.service';
+import { AuthService } from './auth.service';
+import { Sequelize } from 'sequelize-typescript';
+
+@Controller('accounts')
+export class AccountsController {
+  constructor(
+    private readonly accountsService: AccountsService,
+    private readonly authService: AuthService,
+    private readonly sequelize: Sequelize,
+  ) {}
+
+  @Post()
+  async createAccount(
+    @Body() accountData: Accounts,
+  ): Promise<{ data: Accounts; authToken: string }> {
+    const transaction = await this.sequelize.transaction();
+    try {
+      const account = await this.accountsService.insertAccount(
+        accountData,
+        transaction,
+      );
+      const auth = await this.authService.insertAuth(account.id, transaction);
+      const response = {
+        data: account.toJSON(),
+        authToken: auth,
+      };
+      await transaction.commit();
+      return response;
+    } catch (err) {
+      await transaction.rollback();
+      if (err === 'DUPLICATE_ACCOUNT') {
+        throw new HttpException(
+          'Account with provided name already exists',
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+      throw err;
```

Comment:
how does this error proprogate to the client? Is it exposed as ISE or the error object itself?

For example - if sequelize throws an error - will the client get the sequelize error object or an ISE?

If the client gets the sequelize error object - then we need to fix this. We can't expose internal errors to clients - it can contain sensitive informations